### PR TITLE
Fixes #829 - Select an account when not logged in

### LINKED ISSUE #829: Select an account when not logged in
Even when no user is logged into eclipse the message at the top of the Deploy to App Engine Standard Dialog says "Select an account."

There is no account to select in this case. It should say "Login to Google" or something similar.

### PR DESCRIPTION
**Pull request by akerekes**

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/AccountSelector.java [hunk 1]
```diff
@@ -100,6 +100,10 @@ public int selectAccount(String email) {
     return index;
   }
 
+  public boolean isSignedIn() {
+    return loginService.hasAccounts();
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/AccountSelector.java]: To be correct, this should be `getSelectedCredential() != null`. Currently, `AccountSelector` is not being synchronized with `loginService` by its design (#842). The parent of `AccountSelector` also relies on `getSelectedCredential()` eventually.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanelTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    shell = new Shell(Display.getDefault());
+    parent = new Composite(shell, SWT.NONE);
+    when(project.getName()).thenReturn("testProject");
+  }
+
+  @After
+  public void tearDown() {
+    if (parent != null) {
+      parent.dispose();
+    }
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanelTest.java]: I think it's redundant to dispose children (of any level depth) when disposing their parent.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanelTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  @Mock
+  private IProject project;
+  @Mock
+  private IGoogleLoginService loginService;
+  @Mock
+  private Runnable layoutChangedHandler;
+
+  @Before
+  public void setUp() throws Exception {
+    shell = new Shell(Display.getDefault());
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanelTest.java]: Could you add the same TODO for `Shell` from here: https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven.test/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenArchetypeProjectWizardTest.java#L44?

**Comment by akerekes** [comment]
@chanseokoh PTAL

**Pull request merged**